src/Api/Controller/ListUsersController.php

```diff
@@ -69,6 +68,12 @@ protected function data(ServerRequestInterface $request, Document $document)
 
         $actor->assertCan('viewUserList');
 
+        if ($actor->hasPermission('user.viewLastSeenAt')) {
+            // At the moment, only people able to see everyone's last online date can sort by it. Ref #2519
+            // Otherwise this sort field would defeat the privacy setting discloseOnline
+            $this->addSortField('lastSeenAt');
+        }
```

Comment:
This was my alternate implementation (together with re-adding `lastSeenAt` to `public $sortFields` above)

```suggestion
        if (!$actor->hasPermission('user.viewLastSeenAt')) {
            // If a user cannot see everyone's last online date, we prevent them from sorting by it
            // Ideally we want to use the sort, but it needs to take discloseOnline preference into account
            // We use remove instead of add so that extensions can still disable the sort using the extender
            $this->removeSortField('lastSeenAt');
        }
```

Reply:
Ah now I see what you mean. If we're shipping something in core, I lean towards having it on by default more than having it off, so I this method (in the list, alternatively removable) makes sense to me.